### PR TITLE
Implemented fork cards

### LINKED ISSUE #6: Display forks of a given repo
## Solution proposal
- [x] Add a function to get data from the GitHub API using the provided repo
- [x] Add a function that generates a list of cards based on the given data
- [x] Reset wrapper
- [x] Append list of cards to wrapper

### Card content
- [x] Code snippet (markdown)
- [x] Test result (mockup)
- [x] Comment section with a input field
- [x] Radio buttons for selecting ```Done``` ```Review needed``` ```Not reviewed```
- [x] A save button

## Information 
https://api.github.com/users/ruby/ruby/forks

For getting the code to the code snippet this can be used as a base. https://raw.githubusercontent.com/ruby/ruby/master/README.md
And this for syntax highlighting https://highlightjs.org/

### LINKED ISSUE #8: Investigation: Add syntax highlighting to fork cards
## Acceptance criteria
- [x] Investigate how to add syntax highlighting to code snippets. Using markdown. 
- [x] Open an issue with a solution proposal

### LINKED ISSUE #44: Highlight syntax using JS
## Acceptance criteria
- [x] Import hljs
- [x] run provided highlighting

### PR DESCRIPTION
**Pull request by te4-adrian-almetun-smeds**
## What issue did you implement or fix?
Fixed #6
Fixes #8
Fixes #44

## Summary
- Added functions that render the forks of a repo
- Added fetch from github to access code snippets
- Added highlight.js for syntax highlighting 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## How can this implementation be improved?
- Clean up of "callback hell"
- Add comments to js code

### CODE REVIEW

File: public/js/main.js [hunk 1]
```diff
@@ -90,4 +91,47 @@ async function fetchJSON (url) {
...
+async function getCodeSnippet (forkFullName) {
+  forkFullName = 'TE4-oskar-pilborg/smallest_of_two'
+
+  const manifest = await fetchJSON(`https://raw.githubusercontent.com/${forkFullName}/master/.manifest.json`)
+
+  const codeSnippetPromise = await fetch(`https://raw.githubusercontent.com/${forkFullName}/master/${manifest.filePath}`)
+  const codeSnippet = await codeSnippetPromise.text()
+  codeSnippet.trim()
+
+  console.log(codeSnippet)
```
Comment (te4-adrian-almetun-smeds) [inline public/js/main.js]: Is this required?

**Review by te4-andre-skvarc — APPROVED** [review]
This is good work, but it's missing the crucial comments on some functions. So we need to implement these. Still approved! :shipit:

File: public/js/main.js [hunk 2]
```diff
@@ -90,4 +91,47 @@ async function fetchJSON (url) {
...
 async function getRepos (userName) { return await fetchJSON(`/api/github/${userName}/repos`) }
 
-// async function getForks (userRepo) { return await fetchJSON(`/api/github/${userRepo}/forks`) }
+async function getForks (userRepo) { return await fetchJSON(`/api/github/${userRepo}/forks`) }
+
+async function generateForkCards (forkList) {
+  resetWrapper()
+
+  const cardTemplate = cloneTemplate('#forkCardTemplate', '.card')
+
```
Comment (te4-adrian-almetun-smeds) [inline public/js/main.js:100]: I do not think this function name really represent what this function does. Wouldn't it be more appropriate to split this to separate functions or be renamed?

**Review by te4-adrian-almetun-smeds — COMMENTED** [review]
Nicely done overall, just a few pointers.

File: public/js/main.js (hunk 2 above)
Comment (te4-david-jensen) [reply public/js/main.js:100]: Issue has been created to reflect this excessive functionality

**Pull request merged**